### PR TITLE
Trail Map: separate feature list for structure-only variant

### PR DESCRIPTION
**Pull request by aneeshd16**
Related to #90477

## Proposed Changes

* Creates a separate feature list for the structure-only variant. This is needed since we want to customize the order of features in the list and also omit certain features from appearing for a plan. Eg: "Support via email" should not be visible for the Creator plan.
* Also removes `includePreviousPlanFeatures` from everywhere since it is no longer required.
* TODO: There are still some features that are not quite aligned:
  * Free domain
 
  image

  * Growth and monetization tools
  image

#### Full screenshot
localhost_64949_iframe html_args= id=featuresgrid--trail-map-structure viewMode=story

## Testing Instructions

* Go to `/start/plans` and confirm that the features grid and comparison grid are unchanged from production.
* Run `yarn workspace @automattic/plans-grid-next storybook`.
* Check the Trail Map Structure story for the Feature Grid.
* Confirm that the structure variant contains all features from the control variant.

## Pre-merge Checklist

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

### CODE REVIEW

**Comment by oswian** [comment]
The 'Developer tools' section is not showing in the copy + structure variant. Is that intentional?

File: client/my-sites/plans-features-main/index.tsx [hunk 1]
```diff
@@ -331,7 +331,6 @@ const PlansFeaturesMain = ( {
 
 	const {
 		isLoading: isTrailMapExperimentLoading,
-		variant: trailMapExperimentVariant,
```
Comment (oswian) [inline client/my-sites/plans-features-main/index.tsx:334]: As the `variant` value is no longer used, I think we should remove it from the [hook](https://github.com/Automattic/wp-calypso/blob/e9d35d14eab8bdd729d990d1cfd41bbb3c40ae1c/client/my-sites/plans-features-main/hooks/use-experiment-for-trail-map.ts#L48).

**Review by oswian — APPROVED** [review]
Tests well for me.

**Comment by southp** [comment]
I'm going ahead and merging this to catch the timing of launching the test :)

**Pull request merged**